sdk/lib/opentelemetry/sdk/trace/export/batch_span_processor.rb

```diff
@@ -61,7 +62,7 @@ def initialize(exporter:,
             @spans = []
             @pid = nil
             @thread = nil
-            reset_on_fork
+            reset_on_fork(restart_thread: !forking_dispatcher?)
```

Comment:
Note that simply removing this line should lazily start the thread the first time a span is finished.

Reply:
Yep, I'm aware and this is intentional. In general it is desirable to start the BatchSpanProcessor thread at boot time **except** for forking web servers.

Reply:
If you want the option of eagerly or lazily starting the thread, it seems like the simplest change is to pass a `start_export_thread` argument to `initialize`, then:
```suggestion
            reset_on_fork(restart_thread: start_export_thread)
```
We can bikeshed the name of the argument, but this seems a lot less brittle and simpler than the `DispatcherDetector`.